rusk/src/lib/http/chain.rs

```diff
@@ -124,4 +140,72 @@ impl RuskNode {
 
         Ok(ResponseData::new(serde_json::to_value(&info)?))
     }
+
+    /// Calculates various statistics for gas prices of transactions in the
+    /// mempool.
+    ///
+    /// It retrieves a specified number of transactions, sorted by descending
+    /// gas price, and calculates the average, maximum, minimum and median
+    /// prices. If `max_transactions` is not provided, defaults to all
+    /// transactions in the mempool. In the absence of transactions, will
+    /// default to a gas price of 1.
+    ///
+    /// # Arguments
+    /// * `max_transactions` - Optional maximum number of transactions to
+    ///   consider.
+    ///
+    /// # Returns
+    /// A JSON object encapsulating the statistics, or an error if processing
+    /// fails.
+    async fn get_gas_price(
+        &self,
+        max_transactions: Option<usize>,
+    ) -> anyhow::Result<ResponseData> {
+        let max_transactions = max_transactions.unwrap_or(usize::MAX);
+
+        let gas_prices: Vec<u64> =
+            self.db()
+                .read()
+                .await
+                .view(|t| -> anyhow::Result<Vec<u64>> {
+                    Ok(t.get_txs_hashes_sorted_by_fee()?
+                        .take(max_transactions)
+                        .map(|(gas_price, _)| gas_price)
+                        .collect())
+                })?;
+
+        if gas_prices.is_empty() {
+            let stats = serde_json::json!({ "average": 1, "max": 1, "median": 1, "min": 1 });
+            return Ok(ResponseData::new(serde_json::to_value(stats)?));
+        }
+
+        let mean_gas_price = {
+            let total: u64 = gas_prices.iter().sum();
+            let count = gas_prices.len() as u64;
+            // ceiling division to round up
+            (total + count - 1) / count
```

Comment:
Should this by just total/count as usize?

Reply:
It will round down, which means your chance of inclusion would drop. Relatively speaking.